### PR TITLE
Obsoleted `Purchases`.`createAlias`/`identify`/`reset` and simplified `IdentityManager` implementation

### LINKED ISSUE #732: Early return from configure call
Does this logic work?
`if (maybeCurrentAppUserID != nil && maybeAppUserID != nil) && maybeAppUserID != maybeCurrentAppUserID { return }`

We don't know! Currently `maybeCurrentAppUserID` is optional, but we should change that (tracked in #731) , then it becomes a question of:

Is it harmful/helpful/no-op to run

```swift
deviceCache.cache(appUserID: appUserID)
deviceCache.cleanupSubscriberAttributes()
```

If there is no change in `appUserID`?

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes #732. Instead of worrying about whether `IdentityManager.configure` was idempotent or if the implementation was correct, I realized that `Purchases` never called that method after its initialization
Therefore, to simplify the whole implementation, the `configure` part is now in the constructor. Subsequent `logIn`/`logOut` calls are the only way to change the identity.

Some of this code was tangled with some of the deprecations in `IdentityManager`, so to aid in this refactor, I `obsoleted` [3 methods that were already `deprecated` in version 3.13.0](https://github.com/RevenueCat/purchases-ios/blob/3.13.0/Purchases/Public/RCPurchases.h#L205-L224).

This allowed me to remove the associated tests instead of changing them to use the new API. Following #950, these methods are now separate in `Obsoletions.swift`, and they point to the new methods.

The remaining tests in `PurchasesTests` all pass, and the ones in `IdentityManagerTests` have been converted to initializing new `IdentityManager` instances. Those pass as well, so I feel confident in this refactor :muscle:

### CODE REVIEW

**Comment by NachoSoto** [comment]
I removed the calls to these methods from `SwiftAPITester`, but this has exposed an issue: Objective-C is able to call these!
Probably because they're `obsoleted` like this:
```swift
@available(swift, obsoleted: 1, ...)
```

I'm going to fix this for all methods in `Obsoletions.swift`, otherwise Objective-C users can call this with only a warning, but it would result in a `fatalError`!

**Comment by NachoSoto** [comment]
> I removed the calls to these methods from `SwiftAPITester`, but this has exposed an issue: Objective-C is able to call them!

The right solution is to just use `@available(*, unavailable, renamed: "")`, the problem is that looks like there's no way to specify a different new method name for Obj-C and Swift. I haven't had to deal with this in the past (I haven't really touched Obj-C in years). @taquitos / @aboedo any guidance here?

**Comment by aboedo** [comment]
🤔 I think you might still be able to obsolete them by instead of marking `swift`, marking all platforms, i.e.:
```swift
@available(iOS, obsoleted: 1, renamed: "getCustomerInfo(completion:)")
@available(tvOS, obsoleted: 1, renamed: "getCustomerInfo(completion:)")
@available(watchOS, obsoleted: 1, renamed: "getCustomerInfo(completion:)")
@available(macOS, obsoleted: 1, renamed: "getCustomerInfo(completion:)")
@objc func purchaserInfo(completion: @escaping (CustomerInfo?, Error?) -> Void) {
```

**Comment by NachoSoto** [comment]
That's essentially the same as `unavailable`: https://nshipster.com/available/
Since we're making them unavailable for all versions, I think it's more precise.

But regardless, I'm still unsure about the `renamed:` value: should that point to the `Obj-C` parameter name? I'm not sure if Xcode parses that and converts it to Swift automatically?

**Comment by aboedo** [comment]
testing it out locally it seems to correctly identify the objc name, but it doesn't suggest an automatic fix for objc
image

**Comment by NachoSoto** [comment]
So I guess that's always been an issue? Should we just have the `swift` name in `rename:` then?

**Comment by aboedo** [comment]
to clarify, the thing that I tested out locally was specifying all platforms separately instead of `swift`. When I tried with just `swift`, it let me call the method just fine in objc

**Comment by NachoSoto** [comment]
Right, thanks for checking that :)
I'm waiting to see if [somebody on Twitter]() has an answer about providing different names. As to how to obsolete them I don't have a preference :) I just think that 1 line of `@available(*, unavailable)` is simpler than 4, but either way we have the name issue.

**Comment by NachoSoto** [comment]
I fixed this in #981.

**Comment by NachoSoto** [comment]
This branch is ready, but per my conversation with @aboedo, waiting to hear from the support team to make sure it's not too premature to deprecate these methods.

File: Purchases/Misc/Obsoletions.swift [hunk 1]
```diff
@@ -168,6 +168,50 @@ public extension Purchases {
...
+    /**
+     * Resets the Purchases client clearing the saved appUserID.
+     * This will generate a random user id and save it in the cache.
+     */
+    @available(iOS, obsoleted: 1, renamed: "logOut")
+    @available(tvOS, obsoleted: 1, renamed: "logOut")
+    @available(watchOS, obsoleted: 1, renamed: "logOut")
+    @available(macOS, obsoleted: 1, renamed: "logOut")
+    @objc func reset(completion: ((CustomerInfo?, Error?) -> Void)?) {
+        fatalError()
```
Comment (taquitos) [inline Purchases/Misc/Obsoletions.swift:212]: Let's add a message to all these `fatalError()`s like `reset obsoleted, please use logOut`

**Review by taquitos — APPROVED** [review]
Pending support's input, I like this very much 😺

File: Purchases/Misc/Obsoletions.swift (hunk 1 above)
Comment (NachoSoto) [reply Purchases/Misc/Obsoletions.swift:212]: The method is unreachable, it's impossible to call it 😛

**Comment by NachoSoto** [comment]
Just to confirm @taquitos, you're good if we merge this now? :)

**Comment by taquitos** [comment]
@NachoSoto let's merge to help unblock other PRs. We can undo this if we get feedback that suggests it's not the right move.

**Comment by NachoSoto** [comment]
Great thanks!

**Comment by aboedo** [comment]
@NachoSoto @taquitos let's reflect this in our [v4 migration guide in that case](https://github.com/RevenueCat/purchases-ios/blob/main/docs/V4_API_Updates.md)

**Comment by NachoSoto** [comment]
@aboedo added here: https://github.com/RevenueCat/purchases-ios/pull/1010

**Pull request merged**